python/pyspark/errors/exceptions.py

```diff
@@ -0,0 +1,24 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from pyspark.sql.utils import CapturedException
+
+
+class PySparkException(CapturedException):
```

Comment:
As mentioned in PR description, currently all PySpark-specific errors are defined as `PySparkException` class.
It might be necessary to refine the `PySparkException` into multiple categories as the number of PySpark-specific errors increases in the future.

Reply:
Currently, `pyspark.sql.utils` defines some classes to handle the errors from Python worker of Py4J.
I plan to integrate them with `pyspark.errors` package as follow-up when I work on migrating the errors generated by Python worker and Py4J.

Reply:
I think the inheritance hierarchy here is a bit odd since `PySparkException` isn't technically `CapturedException` (captured from Py4J)

Reply:
That makes sense, `PySparkException` should be higher up in the hierarchy than `CapturedException`.

Thanks for catching this! Let me update it.